docs/DeveloperGuide.md

```diff
@@ -1,38 +1,286 @@
-# Developer Guide
+<div align="center">
+<h2> MoneyMind Developer Guide</h2>
+<h3> "Mind your Money" </h3>
+</div>
+
+<div align="left">
 
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Architecture
+
+<img src="images\ArchitectureDiagram.png" width="300" />
+
+The ***Architecture Diagram*** given above explains the high-level design of the App. 
+
+Given below is a quick overview of how each component interact with each other.
+
+The main components in the architecture are:
+
+* `MoneyMind`: The main program of the application, 
+it initializes the other components in the correct sequence and is responsible for shut down the application.
+* `UI`: The user interface of the application.
+* `Storage`: The storage of data of the application.
+* `Data`: The data classes used in the application, including Event, Category and CategoryList.
+* `Command`: The command of the application, including AddCommand, DeleteCommand, ListCommand, etc.
+
+<img src="images\ArchitectureSequence.png" width="500" />
```

Comment:
![image](https://user-images.githubusercontent.com/88263677/229006373-4578ab5d-5442-497d-83d0-64cd6f129cc3.png)
I think its better if the activation bar is unbroken and looks something like this:
![image](https://user-images.githubusercontent.com/88263677/229006488-855f0d7e-b453-431b-a5bf-f24c7598ed24.png)